xds/src/main/java/io/grpc/xds/internal/sds/ClientSslContextProviderFactory.java

```diff
@@ -34,8 +34,8 @@
   private final CertProviderClientSslContextProvider.Factory
       certProviderClientSslContextProviderFactory;
 
-  ClientSslContextProviderFactory() {
-    this(Bootstrapper.getInstance(), CertProviderClientSslContextProvider.Factory.getInstance());
+  ClientSslContextProviderFactory(Bootstrapper bootstrapper) {
```

Comment:
Is this change necessary? For tests you are using the other constructor, but for real instantiation it just calls `Bootstrapper.getInstance()` inside the constructor to get the default implementation. That doesn't sound wrong. It doesn't need to leave this burden to the caller of this class.

Reply:
This is highly desirable. The `Bootstrapper.getInstance()` is now called in only one place: `io.grpc.xds.internal.sds.TlsContextManagerImpl.getInstance()` and that instance is then propagated to all the members. Similar to how the abstract Factories are used wherever possible.

Reply:
It doesn't matter as `Bootstrap.getInstance` returns the singleton.

Reply:
Yes, but it returns an implementation that is fixed and that reads from disk.